### DISCUSSION TITLE
Battery Discharge value too low after H3-12.0-E inverter firmware update

### DISCUSSION THREAD

**Discussion by maddin4711** [nathanmarlor/foxess_modbus#639] [General]
After an inverter firmware update from Master 1.69 to 1.98, the "Battery Discharge" sensor reports values that are too low. The values in the Fox cloud are still ok and correspond: Discharge Power = InvBatVolte x InvBatCurrent.

Example:
InvBatCurrent = 1.1 A
InvBatVoltage = 178 V
Fox Cloud reports Discharge Power = 200 W -> ok
v1.11.1 reports Battery Discharge = 123 W -> too small!

H3-12.0-E inverter Firmware
Master version: 1.98
Slave version: 1.03
Manager version: 1.78

Is this also the case for you, can anyone confirm this?

**Comment by canton7** [comment]
Interesting that it's about 1/2 the expected value. Is it just battery_discharge, or also battery_1_discharge / battery_2_discharge?

**Comment by maddin4711** [comment]
It is just sensor battery_discharge (i have no battery_1_discharge / battery_2). 

I have just observed some very strange behavior.
It seems that the lower discharge power is only displayed at night. This afternoon the power was ok. 

I then installed the new v1.12.1. (HA version still 2024.6.4).
Let's see how it looks next night
![Nighttime_Discharge_low](https://github.com/nathanmarlor/foxess_modbus/assets/104696394/e56992c5-d88d-416f-921b-c873c8d72194)
![Daytime_Discharge_ok](https://github.com/nathanmarlor/foxess_modbus/assets/104696394/9b7508f3-123b-4c61-b61a-302bdf6d1296)

**Reply by canton7** [reply]
Ah H3 not H3-Pro, my bad.

**Reply by maddin4711** [reply]
Yes, it's a H3-12.0-E inverter.

I have also updated HA, so now I have HA version 2024.7.1 and integration v1.12.1, but battery_discharge still seems to be too low at night and even goes to 0W although the battery current is 0.8A and battery voltage is 178.4V

![Nighttime_Discharge_low_2](https://github.com/nathanmarlor/foxess_modbus/assets/104696394/2df3b432-7741-4526-b28f-d8ec5fac59c6)

**Reply by maddin4711** [reply]
...and during daytime it seems to be ok; very curious.

![Daytime_Discharge_ok_2](https://github.com/nathanmarlor/foxess_modbus/assets/104696394/582b66f7-8c9e-43fd-9994-af4d7cd93efa)

**Reply by FozzieUK** [reply]
Hmmm i’ve noticed something similar with the very latest H1 firmware, it seems Fox may be ‘moving’ the inverters internal losses around depending on whether on PV or battery.
My daily losses are now showing almost zero (down from ~ 3kW/day so I seem to have lost my ‘losses’) - may be something to do with load calculation for force charge/discharge work modes but it’s messing up my battery stats and efficiency.
